ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -3602,9 +3626,27 @@ public boolean dropPartition(String dbName, String tableName, List<String> parti
       exprBytes = SerializationUtilities.serializeExpressionToKryo(expr);
     }
     try {
+
       String defaultPartitionName = HiveConf.getVar(conf, ConfVars.DEFAULTPARTITIONNAME);
-      names = getMSC().listPartitionNames(tbl.getCatalogName(), tbl.getDbName(),
-          tbl.getTableName(), defaultPartitionName, exprBytes, order, maxParts);
+
+      if (AcidUtils.isTransactionalTable(tbl)) {
+        PartitionsByExprRequest req =
+            new PartitionsByExprRequest(tbl.getDbName(), tbl.getTableName(), ByteBuffer.wrap(exprBytes));
+        req.setDefaultPartitionName(defaultPartitionName);
+        if (maxParts >= 0) {
+          req.setMaxParts(maxParts);
+        }
+        req.setOrder(order);
+        req.setCatName(tbl.getCatalogName());
+        ValidWriteIdList validWriteIdList = getValidWriteIdList(tbl.getDbName(), tbl.getTableName());
+        req.setValidWriteIdList(validWriteIdList != null ? validWriteIdList.toString() : null);
+        names = getMSC().listPartitionNames(req);
+
+      } else {
+        names = getMSC()
+            .listPartitionNames(tbl.getCatalogName(), tbl.getDbName(), tbl.getTableName(), defaultPartitionName,
```

Comment:
Maybe use the same listPartitionNamesReques for acid and non-acid tables as well?

Reply:
Makes sense, done.